### PR TITLE
refactor: Refactor training pipeline

### PR DESCRIPTION
**Pull request by dataJSA**
This PR aims at refactoring the pyro-risks training pipeline.
Here is the suggestion for the `pyro-risks.models` sub-package and the root modules 

```sh
├── config
│   ├── __init__.py
│   ├── datasets.py
│   ├── logger.py
│   └── models.py
├── models
│   ├── __init__.py
│   ├── pipelines.py
│   ├── transformers.py
│   └── utils.py
├── predict.py
├── train.py
└── version.py
```
The proposed changes aim at reflecting the tools available within the scikit-learn ecosystem here is sklearn documentation on [Dataset transformations](https://scikit-learn.org/stable/data_transforms.html).

- The **transformers**  available within the `sklearn.preprocessing`  package are classes that provide convenient  means of  preprocessing raw datasets. The  custom transformers needed as training steps I propose to define them within `transformers.py` module.

- The Pipeline class can be used to chain multiple transformers and estimators into one, for example chain feature selection, normalization and classification. I proposed to define the different pipelines within the `pipelines.py` module.

I have made a lot of progress so far, here what is left to do:
- [x] write unit tests for the `train.py` module
- [x] update requirements and dependencies
- [x] ensure compatibility with `predict.py`module 
- [x] black formatting

### CODE REVIEW

**Comment by frgfm** [comment]
@dataJSA some mandatory tests seem to be failing, could you take a look at it before we review it please? :pray:

**Comment by dataJSA** [comment]
Yes I flag it as a draft pull request but apparently it notified everybody before I marked it as ready for review 😅. 
I will be ready for review in a few days 🌲

**Comment by frgfm** [comment]
@dataJSA the notifications come because you added people as reviewers ;)

**Comment by dataJSA** [comment]
I just flag the PR as ready despite Codacy Static Code Analysis failing.

I think we should discuss issues related to the different PEP conventions and pylint choices on separate issues. 
Right few rules implemented are mutually exclusive (D203 in conflict and D211, D212 in conflicts with D13).

We need to choose a docstring style and enforce rules based on it, we can open a separate issue as a reminder !

**Review by GHCamille — APPROVED** [review]
Thanks for the PR, really nice job !

Actually your PR seems to have the same Codacy score as the ones we merged before so I guess there is no problem here, right ?

I think you're right about opening an issue about how we can configure Codacy for docstrings !

**Comment by dataJSA** [comment]
Thanks @GHCamille, yes there were no issue related to security or function implementations but only docstrings !

**Comment by frgfm** [comment]
FWI @dataJSA if you don't delete the branch once the PR is merged, you will end up with numerous useless branches ;)
